packages/datetime/src/dateInput.tsx

```diff
@@ -343,6 +343,10 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
      * This is a no-op unless moment-timezone's setDefault has been called.
      */
     private fromDateToMoment = (date: Date) => {
+        if (null == date) {
```

Comment:
check the constructor, where this function is called. i think `defaultValue` and `value` should both check `!= null` instead of whatever they're doing now. that might fix the `null` vs `undefined` distinction